AI/BattleAI/PotentialTargets.cpp

```diff
@@ -69,28 +69,52 @@ PotentialTargets::PotentialTargets(const battle::Unit * attacker, const Hypothet
 		}
 		else
 		{
-			for(BattleHex hex : avHexes)
-				if(CStack::isMeleeAttackPossible(attackerInfo, defender, hex))
-					possibleAttacks.push_back(GenerateAttackInfo(false, hex));
+			for(BattleHex hex : avHexes) {
+				if(!CStack::isMeleeAttackPossible(attackerInfo, defender, hex))
+					continue;
+
+				auto bai = GenerateAttackInfo(false, hex);
+				if(!bai.affectedUnits.empty())
+					possibleAttacks.push_back(bai);
+			}
 
 			if(!vstd::contains_if(possibleAttacks, [=](const AttackPossibility & pa) { return pa.attack.defender->unitId() == defender->unitId(); }))
 				unreachableEnemies.push_back(defender);
 		}
 	}
+
+	boost::sort(possibleAttacks, [](const AttackPossibility & lhs, const AttackPossibility & rhs) -> bool
+	{
+		if(lhs.collateralDamage < rhs.collateralDamage)
+			return false;
+		if(lhs.collateralDamage > rhs.collateralDamage)
+			return true;
+		return (lhs.damageDealt + lhs.shootersBlockedDmg + lhs.damageReceived > rhs.damageDealt + rhs.shootersBlockedDmg + rhs.damageReceived);

```

Comment:
Could you describe me why we add received damage? Don't we want to minimize it instead of maximizing?

Reply:
This is a leftover from the previous iteration where received and collateral damages were negative numbers, which I've changed at DjWarmonger's request. Thanks for spotting this issue.